### ISSUE TITLE
grafana-cli: command not found in Docker which is created from Alpine distro

### ISSUE THREAD

**Issue by tiger-seo** [grafana/grafana#18865]
**What happened**:
- I took branch v6.3.x with latest 6.3.5
- then i've merged fix for error `version 'GLIBC_2.28' not found` using https://github.com/grafana/grafana/pull/18701/files
- then i tried to build my custom grafana image with plugins and have got an error
partial console output:
```
Step 11/11 : RUN if [ ! -z "${GF_INSTALL_PLUGINS}" ]; then     OLDIFS=$IFS;         IFS=',';     for plugin in ${GF_INSTALL_PLUGINS}; do         IFS=$OLDIFS;         grafana-cli --pluginsDir "$GF_PATHS_PLUGINS" plugins install ${plugin};     done; fi
 ---> Running in d54bc918b88b
/bin/sh: 1: grafana-cli: not found
/bin/sh: 1: grafana-cli: not found
/bin/sh: 1: grafana-cli: not found
The command '/bin/sh -c if [ ! -z "${GF_INSTALL_PLUGINS}" ]; then     OLDIFS=$IFS;         IFS=',';     for plugin in ${GF_INSTALL_PLUGINS}; do         IFS=$OLDIFS;         grafana-cli --pluginsDir "$GF_PATHS_PLUGINS" plugins install ${plugin};     done; fi' returned a non-zero code: 127
```
- then i tried to investigate and have added following commands to my custom Dockerfile
```
RUN ls -la /usr/share/grafana/bin/
RUN /usr/share/grafana/bin/grafana-cli
```
and the output was following:
```
Step 11/13 : RUN ls -la /usr/share/grafana/bin/
 ---> Running in 4d34ce298b73
total 65684
drwxr-xr-x 2 root root     4096 Sep  3 15:03 .
drwxr-xr-x 1 root root     4096 Sep  3 15:03 ..
-rwxr-xr-x 1 root root 19709640 Sep  3 15:03 grafana-cli
-rwxr-xr-x 1 root root 47540920 Sep  3 15:03 grafana-server
Removing intermediate container 4d34ce298b73
 ---> a020194082c5
Step 12/13 : RUN /usr/share/grafana/bin/grafana-cli
 ---> Running in 55f5a0742582
/bin/sh: 1: /usr/share/grafana/bin/grafana-cli: not found
The command '/bin/sh -c /usr/share/grafana/bin/grafana-cli' returned a non-zero code: 127
```

**What you expected to happen**:
no error, but successfully built image

**Anything else we need to know?**:
I think it might be related to the fact that grafana image is derived from Alpine distro and there is known issues running some binaries; please, see comment https://github.com/github/hub/issues/1818#issuecomment-409540204 for similar issue

**Comment by xlson** [comment]
I don't quite understand what you have done as everything is working in master while there's no Alpine in `v6.3.x`. My best guess would be that you have only merged part of what you need to make it work. If you really want to run 6.3.5 in Docker on alpine I would suggest taking the complete `Dockerfile` from master and bringing that back into your v6.3.x branch.

**Comment by tiger-seo** [comment]
@xlson when i'm trying to build `v6.3.x` then i get the error `version 'GLIBC_2.28' not found`; and this error is not yet fixed in `v6.3.x`, but fixed for master in https://github.com/grafana/grafana/pull/18701/

**Comment by xlson** [comment]
If you get that error when trying to build the `v6.3.x` branch it means you have either merged some other change that causes it, or that there's something with you environment causing the issue. The PR you are referencing doesn't make sense for the `v6.3.x` branch as it solves a problem that hasn't been introduced in that branch.

If your fork of Grafana is public you can link it and I can take a look to try to figure out why it isn't working. If not, I will need more detail about what you have done. A copy of the `Dockerfile` you use to build Grafana might do the trick.

**Comment by tiger-seo** [comment]
@xlson 
grafana image is built from https://github.com/tiger-seo/grafana/tree/v6.3.x
```
FROM tigerseo/grafana

USER root
RUN apt-get update && apt-get install -y wget
ENV GF_PATHS_PLUGINS=/grafana/plugins
RUN mkdir -p "$GF_PATHS_PLUGINS"
RUN chown grafana -R "$GF_PATHS_PLUGINS"
USER grafana

ARG GF_INSTALL_PLUGINS="simpod-json-datasource,grafana-piechart-panel"

RUN if [ ! -z "${GF_INSTALL_PLUGINS}" ]; then \
    OLDIFS=$IFS; \
        IFS=','; \
    for plugin in ${GF_INSTALL_PLUGINS}; do \
        IFS=$OLDIFS; \
        grafana-cli --pluginsDir "$GF_PATHS_PLUGINS" plugins install ${plugin}; \
    done; \
fi
```

**Comment by AndrzejKasiewicz** [comment]
I confirm the issue with building v6.3.5 on Docker. I'm building it on macOS by using:
`docker build -t grafana/grafana:6.3.5 .`

Image is successfully built but running it throws:
`grafana-server: /lib/x86_64-linux-gnu/libc.so.6: version GLIBC_2.28 not found (required by grafana-server)` 

I tried building v6.3.x branch with Dockerfile copied from master and it works.

**Comment by marefr** [comment]
@tiger-seo https://github.com/tiger-seo/grafana/blob/v6.3.x/Dockerfile#L36 and grafana/6.3.x branch are using Ubuntu 18:04 as base image. 

Are you both using the Dockerfile in root?

**Comment by AndrzejKasiewicz** [comment]
@marefr 
Yes, I used Dockerfile from the root directory from 6.3.x branch. Please let me know if I can help you with finding a way to fix it. Or it can be ignored as problem doesn't occur on 6.4 as it uses alpine.

**Comment by marefr** [comment]
Not many has reported this so would suggest using 6.4 if possible instead.

**Comment by tiger-seo** [comment]
@marefr yes, the Dockerfile from the root;  also, if i change the base go image to its previous value `golang:1.12.4` - the problem is gone, so clearly it is something wrong with the `golang:1.12.9`

**Comment by Maxattax97** [comment]
I ran into this thread because `grafana-cli` and `grafana-server` were missing from my container. I know my problem is not the same as yours (you've got compiler issues going on), but I want to document for anyone else who may run into this problem and who sees this thread: _Do not make a volume mount for the Grafana home directory `/usr/share/grafana`._ Dumb move on my part, led to lots of fruitless debugging.

Carry on, and good luck :+1:

**Comment by marefr** [comment]
@AndrzejKasiewicz @tiger-seo can confirm I get the same error as you when building and running the v6.3.x docker image.

**Comment by cflee** [comment]
I just ran into this issue trying to build v6.3.5 using the Dockerfile in root directory. 

It seems the issue is around 1.12.7 onwards, the default golang image was switched from Debian Stretch to Buster, and Buster uses glibc 2.28 which isn't in Ubuntu 18.04 -- see issue docker-library/golang#298 which points to PR docker-library/golang#290.

If the first stage in v6.3.x Dockerfile is changed from `golang:1.12.9` to `golang:1.12.9-stretch`, the build seems to be ok.

**Comment by marefr** [comment]
@cflee thank you, appreciate it. We would accept a PR targeting v6.3.x branch if anyone is up for it.

**Comment by marefr** [comment]
Fixed by #19394
